raster-foundry/raster-foundry
# Modal redesign

## Overview

- Initial front-end revamp of modals for the platform
- This PR completes the redesign of the following modals: 
    - `Publishing options` modal for projects
    - `Project deletion` modals for projects
    - `New organization` creation modals within platform management
    - `New team` creation modals within team management (not deletion modal)
- The following modals needed low-level fixes to maintain usability as we update modals:
    - `Permission` modal
- Removes the need for theme `_modal.scss` overrides

### Checklist
I don't believe this PR requires any of these checklist items:

~- [ ] Styleguide updated, if necessary~
~- [ ] [Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~
~- [ ] PR has a name that won't get you publicly shamed for vagueness~
~- [ ] Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
~- [ ] Any new SQL strings have tests~

### Demo

**New team modal**
<img width="1067" alt="screen shot 2018-09-28 at 4 22 45 pm" src="https://user-images.githubusercontent.com/1928955/46232114-1b672a00-c33c-11e8-95c5-e7111c474e4c.png">

**New organization modal**
<img width="1070" alt="screen shot 2018-09-28 at 4 23 37 pm" src="https://user-images.githubusercontent.com/1928955/46232115-1b672a00-c33c-11e8-8c69-b35f0fba74c6.png">

**Project publication options modal** also this applies to lab node sharing?
<img width="1058" alt="screen shot 2018-09-28 at 12 47 53 pm" src="https://user-images.githubusercontent.com/1928955/46232132-3043bd80-c33c-11e8-9f68-57c811ea2789.png">

**Project deletion modal**
<img width="1065" alt="screen shot 2018-09-28 at 12 48 05 pm" src="https://user-images.githubusercontent.com/1928955/46232133-3043bd80-c33c-11e8-86a1-fb86dd020750.png">

**Temporary fix to permissions modal**
<img width="1059" alt="screen shot 2018-09-28 at 12 48 15 pm" src="https://user-images.githubusercontent.com/1928955/46232134-3043bd80-c33c-11e8-9c79-bb0c563f4b24.png">


### Notes
- All three themes have been tested, but feel free to check them.
- Currently jenkins is failing due to some comma issues in `/app-frontend/src/app/pages/admin/organization/teams/teams.module.js` and `/Users/mwilliams/dev/work/raster-foundry/app-frontend/src/app/pages/admin/platform/organizations/organizations.module.js`. I couldn't figure out how to fix so could use some support on that front.


## Testing Instructions

 * Test as many modals as you can to make sure they are all usable still.
 * Check modals highlighted above for consistancy with designs in [Abstract](https://share.goabstract.com/457b1711-5d3f-4f5b-a1dc-88df6ca567be)

This starts the work on #2910, #3843
